dtnd crashes when trying to dispatch a deleted bundle

Apparently, it can happen that a bundle is deleted from the store while it is still in the dispatching pipeline. Then when the routing algorithm tries to access the bundle's data, dtnd crashes.

Example log:
```
INFO[11:05:52.475] Store deletes BundleItem                      bundle="dtn://node-1/-619002318-0"
INFO[11:05:52.476] Retrying bundle from store                    bundle="dtn://node-1/-619002318-0"
INFO[11:05:52.481] Dispatching bundle                            bundle="dtn://node-1/-619002318-0"
panic: No data found for this key

goroutine 6601 [running]:
github.com/dtn7/dtn7-go/core.(*BundlePack).MustBundle(...)
	/home/msommer/devel/dtn7-go/core/bundle_pack.go:120
github.com/dtn7/dtn7-go/core.(*Core).dispatching(0xc000166120, 0x1, 0x9ad640, 0xc0016155e0, 0x24e539ce, 0x0, 0x0, 0x0, 0x0, 0x1, ...)
	/home/msommer/devel/dtn7-go/core/processing.go:140 +0x621
github.com/dtn7/dtn7-go/core.(*Core).checkPendingBundles(0xc000166120)
	/home/msommer/devel/dtn7-go/core/core.go:105 +0x304
created by github.com/dtn7/dtn7-go/core.(*Cron).fire
	/home/msommer/devel/dtn7-go/core/cron.go:65 +0x1cb

```

--- a/core/processing.go
+++ b/core/processing.go
@@ -137,7 +137,16 @@ func (c *Core) dispatching(bp BundlePack) {
 		return
 	}
 
-	if c.HasEndpoint(bp.MustBundle().PrimaryBlock.Destination) {
+	bndl, err := bp.Bundle()
+	if err != nil {
+		log.WithFields(log.Fields{
+			"bundleID": bp.Id,
+			"error":    err.Error(),
+		}).Warn("Error dispatching bundle")
+		return
+	}
+
+	if c.HasEndpoint(bndl.PrimaryBlock.Destination) {
 		c.localDelivery(bp)
 	} else {
 		c.forward(bp)

